lib/upnp/igd_service.go

```diff
@@ -50,6 +50,39 @@ type IGDService struct {
 	URL       string
 	URN       string
 	LocalIP   net.IP
+	PinholeID uint16
+}
+
+// TryAddPinhole adds an IPv6 pinhole in accordance to http://upnp.org/specs/gw/UPnP-gw-WANIPv6FirewallControl-v1-Service.pdf
+// We just use the same external and internal port
+func (s *IGDService) TryAddPinhole(ctx context.Context, protocol nat.Protocol, port int, description string, duration time.Duration) (int, error) {
+	var protoNumber int
+	if protocol == nat.TCP {
+		protoNumber = 6
+	} else if protocol == nat.UDP {
+		protoNumber = 17
+	}
+
+	tpl := `<u:AddPinhole xmlns:u="%s">
```

Comment:
Global const?

Reply:
Why make it a global constant? The template is only relevant to the method it's used in? It could be a local constant though

Reply:
I've made it const now